lti/lti-common/src/java/org/sakaiproject/lti13/LineItemUtil.java

```diff
@@ -120,13 +120,16 @@ public static String constructExternalId(Map<String, Object> content, SakaiLineI
 	 */
 	// TODO: Remember to dream that someday this will be JSON :)
 	public static String constructExternalId(Long tool_id, Map<String, Object> content, SakaiLineItem lineItem)
+	{
+		Long content_id = (content == null) ? null : ((Integer) content.get(LTIService.LTI_ID)).longValue();
```

Comment:
```suggestion
		Long content_id = (content == null) ? null : LTI13Util.getLongNull(content.get(LTIService.LTI_ID));
```
@csev Also, at the top of the source file, you'll need to import org.tsugi.lti13.LTI13Util. I've tested this suggested change with McGrawHill, and it works. It's adapted from the Oracle issue raised by @jesusmmp .